src/readbin/symbols.ml

```diff
@@ -66,4 +66,21 @@ let read ?demangle ~filename arch base  : string table =
             let mem = Memory.view ~from ~words base |> ok_exn in
             let name = demangle name in
             Table.add syms mem name |> ok_exn
-          with exn -> syms))
+          with _exn -> syms))
+
+let read_addrset filename : Addr.Set.t =
+  Addr.Set.t_of_sexp @@ Sexp.load_sexp filename
+
+let write_addrset ?filename (addr_set : Addr.Set.t) : unit =
+  match filename with
+  | None -> Sexp.output Out_channel.stdout @@ Addr.Set.sexp_of_t addr_set
+  | Some f -> (
+      let oc = Out_channel.create f in
```

Comment:
There is a safe software pattern that captures "open/do-work/close" stuff. For files in OCaml there is `with_file` function, that takes a filename and a user function, opens the file, applies the function to the file descriptor, and then safely closes `fd`, even in a case of error.